### PR TITLE
[Login Frontend] Introducing the registration flow

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

This PR is introducing the registration flow, meaning that we are now able to create new users from the registration form and login them.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Registration Form:**

image

**Creating a new account:**

image

**Account Created:**

image

**If the user (email address) already exists in the system:**

image

**Login in with the new user credentials:**

image

image

# Testing

Manual steps to reproduce this functionality:

1.  Create a new account by using the registration form.
2.  Make sure that you are able to login once the new account has been created.
3.  Try to create the exact same account and make sure that you receive the _already existing account_ error.

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)~~
- [x] My changes look good on IE 11+ and Chrome

### CODE REVIEW

File: frontend/src/components/authentication/RegistrationForm.jsx [hunk 1]
```diff
@@ -305,9 +357,35 @@ class CreateAccountForm extends Component {
...
         </div>
+        <PopupBox
+          isCloseButtonVisible={false}
+          isBackdropStatic={true}
+          show={this.state.showDialog}
+          handleClose={this.redirectToLoginPage}
+          title={"Account Created"}
+          description={
+            <div>
+              <p>{"You have just created a new account!"}</p>
```
Comment (caleybrock) [inline frontend/src/components/authentication/RegistrationForm.jsx]: These should be localized.

File: frontend/src/components/authentication/RegistrationForm.jsx [hunk 2]
```diff
@@ -110,6 +130,33 @@ class CreateAccountForm extends Component {
...
+  redirectToLoginPage = () => {
+    // refresh the page in order to show the login form
+    window.location.reload();
+    // close dialog
+    this.setState({ showDialog: false });
+  };
+
+  handleSubmit = event => {
+    this.props
+      .registerAction({
```
Comment (caleybrock) [inline frontend/src/components/authentication/RegistrationForm.jsx]: This would be a good time to ensure this redux action is tested.

**Comment by caleybrock** [comment]
Given our current deploy / master state all of these are save to merge (with normal review and testing process).

File: frontend/src/components/authentication/RegistrationForm.jsx (hunk 2 above)
Comment (fnormand01) [reply frontend/src/components/authentication/RegistrationForm.jsx]: This registration API will be tested in our API tests.
See PR #299.

**Pull request merged**